### ISSUE TITLE
Return focus to organization state dropdown after closing unsupported STLT modal

### ISSUE THREAD

**Issue by mpbrown** [CDCgov/prime-simplereport#9388]
## Area or Component

Organization sign up flow in `OrganizationForm`

## Quick Summary

When a user selects a STLT that is not supported by SR, a modal pops up informing them that they can join the waitlist. When the modal is closed, focus does not return to the STLT dropdown.

### How to Replicate

- Navigate to the sign-up flow and select "My organization is new to SimpleReport"
- Use keyboard to navigate to "Organization state" dropdown
- Use keyboard to select an unsupported state like WV
- Close the modal using the keyboard
- Use tab to observe the keyboard focus does not return to the dropdown automatically

## Accessibility Violation

[When the modal dialog or similar element is closed, keyboard focus is not returned to the triggering element.](https://docs.deque.com/issue-help/1.0.0/en/focus-modal-not-returned)

## Action Requested

Return focus back to STLT dropdown after the modal closes.

Now that we've transitioned from ReportStream to AIMS, we should probably just remove any behavior in the app related to unsupported STLTs since we should be able to support all STLTs through AIMS.

## Screenshots (optional)

<img width="722" height="270" alt="Image" src="https://github.com/user-attachments/assets/28d3bfb3-d649-4505-9de8-266437058ef4" />

## Original Deque Issue
https://cdc-devtools.dequecloud.com/issues/bfe6bee2-5acd-481e-9041-aa66ae591ff4
